lib/src/default_index/readonly.rs

```diff
@@ -156,9 +169,14 @@ impl CommitLookupEntry<'_> {
 /// u32: number of overflow parent entries
 /// for each entry, in some topological order with parents first:
 ///   u32: generation number
-///   u32: number of parents
-///   u32: global index position for parent 1
-///   u32: position in the overflow table of parent 2
+///   if number of parents <= 2:
+///     u32: (< 0x8000_0000) global index position for parent 1
+///          (==0xffff_ffff) no parent 1
+///     u32: (< 0x8000_0000) global index position for parent 2
+///          (==0xffff_ffff) no parent 2
+///   else:
+///     u32: (>=0x8000_0000) position in the overflow table, negated
+///     u32: (>=0x8000_0000) number of parents (in the overflow table), negated
 ///   <change id length number of bytes>: change id
```

Comment:
nit: "negated" in the description made me think it was the two's complement. Maybe "bit-negated" or "bit-flipped" or something.

Actually, would it work equally well to just flip the MSB? That would make these values a bit easier to debug if we ever need to do that. I guess that would make the value for "no parent 1/2" be 0x8000_000 to indicate that there are 0 parents starting a position 0. Up to you.

Reply:
> "bit-negated"

+1

> would it work equally well to just flip the MSB? That would make these values a bit easier to debug.

Yes, it also works. Actually I chose the bit-negated representation because `ffff ffff` is significant in hex dump (and so I feel easy to debug.) I don't have strong preference, though.

Reply:
I'll go with the bit-negated representation as there seems no strong preference.